src/pybamm/util.py

```diff
@@ -108,38 +108,77 @@ def __getitem__(self, key):
                 f"'{key}' not found. Best matches are {best_matches}"
             ) from error
 
-    def search(self, key, print_values=False):
+    def search(self, keys, print_values=False):
         """
-        Search dictionary for keys containing 'key'. If print_values is True, then
-        both the keys and values will be printed. Otherwise just the values will
-        be printed. If no results are found, the best matches are printed.
+        Search dictionary for keys containing all terms in 'keys'.
+        If print_values is True, both the keys and values will be printed.
+        Otherwise, just the keys will be printed. If no results are found,
+        the best matches are printed.
+
+        Parameters
+        ----------
+        keys : str or list of str
+            Search term(s)
+        print_values : bool, optional
+            If True, print both keys and values. Otherwise, print only keys.
+            Default is False.
         """
-        key_in = key
-        key = key_in.lower()
 
-        # Sort the keys so results are stored in alphabetical order
-        keys = list(self.keys())
-        keys.sort()
-        results = {}
+        if not keys or (isinstance(keys, str) and not keys.strip()):
+            print("Error: Please provide a non-empty search term.")
```

Comment:
Question: How about we raise actual `ValueError` here? Since we're anyways exiting after this

Reply:
I skipped it earlier because we didn't have a return value in the function. On second thought, yes, raising an error instead of `print("Error: <...>")` sounds good to me.

Reply:
This should be a `TypeError`. The wording can also be better -
```py
err = f"keys must be a list or list of strings; got {type(keys)}"
raise TypeError(err)
```

Reply:
Ah yes, thanks for catching that! The test also needs to be updated accordingly to catch this error.